src/action_executor.cpp

```diff
@@ -138,12 +141,27 @@ bool ActionExecutor::IsDone(std::string* error) const {
           *error = errors::kNoLandmarksDetected;
         }
         world_->surface_box_landmarks.clear();
-        for (size_t i=0; i<result->landmarks.size(); ++i) {
+        for (size_t i = 0; i < result->landmarks.size(); ++i) {
           msgs::Landmark landmark;
           ProcessSurfaceBox(result->landmarks[i], &landmark);
           world_->surface_box_landmarks.push_back(landmark);
         }
         runtime_viz_.PublishSurfaceBoxes(world_->surface_box_landmarks);
+
+        msgs::Surface surface = result->surface;
+        shape_msgs::SolidPrimitive surface_shape;
+        surface_shape.dimensions.resize(3);
+        surface_shape.dimensions[0] = surface.dimensions.x;
+        surface_shape.dimensions[1] = surface.dimensions.y;
+        surface_shape.dimensions[2] = surface.dimensions.z;
+
+        moveit_msgs::CollisionObject surface_obj;
+        surface_obj.header.frame_id = robot_config_.base_link();
```

Comment:
The frame ID should be `surface.pose_stamped.header.frame_id`. Notice that the surface's pose is given as a `PoseStamped`, this means that `surface.pose_stamped.pose`, used below, *must* be interpreted in the frame given in `surface.pose_stamped.header.frame_id`.